### PR TITLE
Update MinimumSumPartition.java

### PR DESCRIPTION
**Pull request by tribbleofjim**
fix code issue

### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**

* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Java files are placed inside an existing directory.
* [ ] All filenames are in all uppercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Java naming conventions.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

**Comment by tribbleofjim** [comment]
It's my first time to commit a file. I want to explain why I did that.
In the origin code, if we use array[1, 2, 3, 9] to test it, we will see that the code return an answer 5 instead of the correct answer 3.(subset{1, 2, 3} and subset{9})
so I fixed it.

File: DynamicProgramming/MinimumSumPartition.java [hunk 1]
```diff
@@ -20,56 +20,73 @@
...
-	    	{
-	        	if(dp[n][i] == true)
-	            		index[p++] = i;
-	    	}
-	    	return index;
-	 }
+	public static int subSet(int arr[]){
+        int n = arr.length;
+        int sum = getSum(arr);
+        boolean dp[][] = new boolean[n+1][sum+1];
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:26]:
```suggestion
        boolean[][] dp = new boolean[n + 1][sum + 1];
```

File: DynamicProgramming/MinimumSumPartition.java [hunk 2]
```diff
@@ -20,56 +20,73 @@
...
+        for(int i = 0; i < n; i++){
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:27]:
```suggestion
        for (int i = 0; i <= n; i++) {
```

File: DynamicProgramming/MinimumSumPartition.java [hunk 3]
```diff
@@ -20,56 +20,73 @@
...
+            dp[i][0] = true;
+        }
+        for(int j = 0; j < sum; j++){
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:30]:
```suggestion
        for (int j = 0; j <= sum; j++) {
```

File: DynamicProgramming/MinimumSumPartition.java [hunk 4]
```diff
@@ -20,56 +20,73 @@
...
+                    dp[i][j] = dp[i-1][j - arr[i-1]] || dp[i-1][j];
+                }
+                else if(arr[i-1] == j){
+                    dp[i][j] = true;
+                }
+                else{
+                    dp[i][j] = dp[i-1][j];
+                }
+            }
+        }
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:47]:
```suggestion
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= sum; j++) {
                if (arr[i - 1] < j) {
                    dp[i][j] = dp[i - 1][j - arr[i - 1]] || dp[i - 1][j];
                } else if (arr[i - 1] == j) {
                    dp[i][j] = true;
                } else {
                    dp[i][j] = dp[i - 1][j];
                }
            }
        }
```

File: DynamicProgramming/MinimumSumPartition.java [hunk 5]
```diff
@@ -20,56 +20,73 @@
...
+
+        // fill the index array
+        int index[] = new int[sum];
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:50]:
```suggestion
        int[] index = new int[sum];
```

File: DynamicProgramming/MinimumSumPartition.java [hunk 6]
```diff
@@ -20,56 +20,73 @@
...
+        int p = 0;
+        for(int i = 0; i <= sum / 2; i++){
+            if(dp[n][i]){
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:53]:
```suggestion
        for (int i = 0; i <= sum / 2; i++) {
            if (dp[n][i]) {
```

File: DynamicProgramming/MinimumSumPartition.java [hunk 7]
```diff
@@ -20,56 +20,73 @@
...
+    public static int getSum(int arr[]){
+        if(arr.length <= 0){
+            return 0;
+        }
+        int sum = 0;
+        for(int i = 0; i < arr.length; i++){
+            sum += arr[i];
+        }
+        return sum;
+    }
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:70]:
```suggestion
    /**
     * Calculate sum of array elements
     *
     * @param arr the array
     * @return sum of given array
     */
    public static int getSum(int[] arr) {
        int sum = 0;
        for (int temp : arr) {
            sum += temp;
        }
        return sum;
    }
```

File: DynamicProgramming/MinimumSumPartition.java [hunk 8]
```diff
@@ -20,56 +20,73 @@
...
+    public static int getMin(int arr[], int sum){
+        if(arr.length <= 0){
+            return 0;
+        }
+        int min = Integer.MAX_VALUE;
+        for(int i = 0; i < arr.length; i++){
+            min = Math.min(min, (sum - 2*arr[i]));
+        }
+        return min;
+    }
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:81]:
```suggestion
    public static int getMin(int[] arr, int sum) {
        if (arr.length == 0) {
            return 0;
        }
        int min = Integer.MAX_VALUE;
        for (int temp : arr) {
            min = Math.min(min, sum - 2 * temp);
        }
        return min;
    }
```

File: DynamicProgramming/MinimumSumPartition.java [hunk 9]
```diff
@@ -20,56 +20,73 @@
...
+
+    public static void main(String args[]){
+        Scanner in = new Scanner(System.in);
+        int n = in.nextInt();
+        int arr[] = new int[n];
+        for(int i = 0 ; i<n; i++){
+            arr[i] = in.nextInt();
+        }
+        System.out.println(subSet(arr));
+    }
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:91]:
```suggestion
    /**
     * Driver Code
     */
    public static void main(String[] args) {
        assert subSet(new int[]{1, 6, 11,5}) == 1;
        assert subSet(new int[]{36, 7, 46, 40}) == 23;
        assert subSet(new int[]{1, 2, 3, 9}) == 3;
    }
```

**Review by shellhub — CHANGES_REQUESTED** [review]
Please use java style instead of c-style. And format code with your IDE (e.g. IntelliJ IDEA) . Thanks :)

File: DynamicProgramming/MinimumSumPartition.java [hunk 10]
```diff
@@ -20,56 +20,73 @@
...
-	    	int index[] = new int[sum];
-	    	int p = 0;
-	    	for(int i = 0 ; i <= sum / 2; i++)
-	    	{
-	        	if(dp[n][i] == true)
-	            		index[p++] = i;
-	    	}
-	    	return index;
-	 }
+	public static int subSet(int arr[]){
```
Comment (shellhub) [inline DynamicProgramming/MinimumSumPartition.java:23]:
```suggestion
	public static int subSet(int[] arr) {
```

**Review by shellhub — CHANGES_REQUESTED** [review]

**Review by tribbleofjim — COMMENTED** [review]
I have resolved these changes. Thanks a lot for the advisor. :)

**Review by shellhub — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**